[Bug]: OCR error after 390+ pages

### Installation Method

Local Installation

### The Problem

I have some documents with many pages 300+, and need to do OCR on them, my PC can handle that kind of process, but reaching page 397, i get this error in the logs bellow, from the logs it's saying that the file was being used by another process. but it is the temp file not the original PDF

### Version of Stirling-PDF

2.14.1

### Last Working Version of Stirling-PDF

_No response_

### Page Where the Problem Occurred

_No response_

### Docker Configuration

```txt

```

### Relevant Log Output

```txt
2026-07-08 18:40:59,979 WARN s.s.c.s.TempFileCleanupService [scheduled-vt-2] Failed to clean up temporary directory: C:\Users\PEDROB~1\AppData\Local\Temp\stirling-pdf\stirling-pdf-11610625914526437198
java.nio.file.FileSystemException: C:\Users\PEDROB~1\AppData\Local\Temp\stirling-pdf\stirling-pdf-11610625914526437198\output\page_397.pdf: O arquivo já está sendo usado por outro processo
	at java.base/sun.nio.fs.WindowsException.translateToIOException(Unknown Source)
	at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(Unknown Source)
	at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(Unknown Source)
	at java.base/sun.nio.fs.WindowsFileSystemProvider.implDelete(Unknown Source)
	at java.base/sun.nio.fs.AbstractFileSystemProvider.deleteIfExists(Unknown Source)
	at java.base/java.nio.file.Files.deleteIfExists(Unknown Source)
	at stirling.software.common.util.GeneralUtils$1.visitFile(GeneralUtils.java:220)
	at stirling.software.common.util.GeneralUtils$1.visitFile(GeneralUtils.java:216)
	at java.base/java.nio.file.Files.walkFileTree(Unknown Source)
	at java.base/java.nio.file.Files.walkFileTree(Unknown Source)
	at stirling.software.common.util.GeneralUtils.deleteDirectory(GeneralUtils.java:214)
	at stirling.software.common.service.TempFileCleanupService.scheduledCleanup(TempFileCleanupService.java:148)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.springframework.scheduling.support.ScheduledMethodRunnable.runInternal(ScheduledMethodRunnable.java:128)
	at org.springframework.scheduling.support.ScheduledMethodRunnable.lambda$run$1(ScheduledMethodRunnable.java:122)
	at io.micrometer.observation.Observation.observe(Observation.java:569)
	at org.springframework.scheduling.support.ScheduledMethodRunnable.run(ScheduledMethodRunnable.java:122)
	at org.springframework.scheduling.config.Task$OutcomeTrackingRunnable.run(Task.java:88)
	at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.runAndReset(Unknown Source)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.VirtualThread.run(Unknown Source)
2026-07-08 18:41:00,313 INFO s.s.c.u.ProcessExecutor [virtual-12086] Estimating resolution as 493
2026-07-08 18:41:00,313 INFO s.s.c.u.ProcessExecutor [virtual-12086] 
2026-07-08 18:41:00,313 INFO s.s.c.u.ProcessExecutor [virtual-12086] Error in fopenReadStream: failed to open locally with tail page_397.png for filename C:\Users\PEDROB~1\AppData\Local\Temp\stirling-pdf\stirling-pdf-11610625914526437198\images\page_397.png
2026-07-08 18:41:00,313 INFO s.s.c.u.ProcessExecutor [virtual-12086] 
2026-07-08 18:41:00,313 INFO s.s.c.u.ProcessExecutor [virtual-12086] Leptonica Error in findFileFormat: image file not found: C:\Users\PEDROB~1\AppData\Local\Temp\stirling-pdf\stirling-pdf-11610625914526437198\images\page_397.png
2026-07-08 18:41:00,313 INFO s.s.c.u.ProcessExecutor [virtual-12086] 
2026-07-08 18:41:00,939 ERROR s.s.c.a.AutoJobAspect [virtual-10108] AutoJobAspect caught exception during job execution: Can't create an ImageOutputStream!
javax.imageio.IIOException: Can't create an ImageOutputStream!
	at java.desktop/javax.imageio.ImageIO.write(Unknown Source)
	at stirling.software.SPDF.controller.api.misc.OCRController.processWithTesseract(OCRController.java:394)
	at stirling.software.SPDF.controller.api.misc.OCRController.processPdfWithOCR(OCRController.java:157)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:359)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:190)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:158)
	at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:94)
	at stirling.software.common.aop.AutoJobAspect.lambda$wrapWithJobExecution$0(AutoJobAspect.java:83)
	at stirling.software.common.aop.AutoJobAspect.lambda$wrapWithMDC$0(AutoJobAspect.java:318)
	at stirling.software.common.service.JobExecutorService.lambda$runJobGeneric$3(JobExecutorService.java:182)
	at stirling.software.common.service.JobExecutorService.lambda$executeWithTimeout$0(JobExecutorService.java:441)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPerTaskExecutor$TaskRunner.run(Unknown Source)
	at java.base/java.lang.VirtualThread.run(Unknown Source)
2026-07-08 18:41:00,940 ERROR s.s.c.s.JobExecutorService [jetty-5230] Error executing synchronous job: javax.imageio.IIOException: Can't create an ImageOutputStream!
java.lang.RuntimeException: javax.imageio.IIOException: Can't create an ImageOutputStream!
	at stirling.software.common.aop.AutoJobAspect.lambda$wrapWithJobExecution$0(AutoJobAspect.java:95)
	at stirling.software.common.aop.AutoJobAspect.lambda$wrapWithMDC$0(AutoJobAspect.java:318)
	at stirling.software.common.service.JobExecutorService.lambda$runJobGeneric$3(JobExecutorService.java:182)
	at stirling.software.common.service.JobExecutorService.lambda$executeWithTimeout$0(JobExecutorService.java:441)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPerTaskExecutor$TaskRunner.run(Unknown Source)
	at java.base/java.lang.VirtualThread.run(Unknown Source)
Caused by: javax.imageio.IIOException: Can't create an ImageOutputStream!
	at java.desktop/javax.imageio.ImageIO.write(Unknown Source)
	at stirling.software.SPDF.controller.api.misc.OCRController.processWithTesseract(OCRController.java:394)
	at stirling.software.SPDF.controller.api.misc.OCRController.processPdfWithOCR(OCRController.java:157)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:359)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:190)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:158)
	at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:94)
	at stirling.software.common.aop.AutoJobAspect.lambda$wrapWithJobExecution$0(AutoJobAspect.java:83)
	... 6 common frames omitted
```

### Additional Information

_No response_

### Browsers Affected

_No response_

### No Duplicate of the Issue

- [x] I have verified that there are no existing issues raised related to my problem.

--- a/app/common/src/main/java/stirling/software/common/service/TempFileCleanupService.java
+++ b/app/common/src/main/java/stirling/software/common/service/TempFileCleanupService.java
@@ -144,8 +144,10 @@ public class TempFileCleanupService {
         int directoriesDeletedCount = 0;
         for (Path directory : registry.getTempDirectories()) {
             try {
-                if (Files.exists(directory)) {
+                if (Files.exists(directory)
+                        && shouldDeleteRegisteredDirectory(directory, maxAgeMillis)) {
                     GeneralUtils.deleteDirectory(directory);
+                    registry.unregisterDirectory(directory);
                     directoriesDeletedCount++;
                     log.debug("Cleaned up temporary directory: {}", directory);
                 }
@@ -273,6 +275,21 @@ public class TempFileCleanupService {
         }
 
         return totalDeletedCount.get();
+    }
+
+    private boolean shouldDeleteRegisteredDirectory(Path directory, long maxAgeMillis) {
+        if (maxAgeMillis <= 0) {
+            return true;
+        }
+
+        try {
+            long currentTime = System.currentTimeMillis();
+            long lastModified = Files.getLastModifiedTime(directory).toMillis();
+            return (currentTime - lastModified) > maxAgeMillis;
+        } catch (IOException e) {
+            log.debug("Could not check directory age, skipping cleanup: {}", directory, e);
+            return false;
+        }
     }
 
     /** Get the system temp directory path based on configuration or system property. */

--- a/app/common/src/main/java/stirling/software/common/util/TempFileManager.java
+++ b/app/common/src/main/java/stirling/software/common/util/TempFileManager.java
@@ -155,6 +155,7 @@ public class TempFileManager {
         if (directory != null && Files.isDirectory(directory)) {
             try {
                 GeneralUtils.deleteDirectory(directory);
+                registry.unregisterDirectory(directory);
                 log.debug("Deleted temp directory: {}", directory.toString());
             } catch (IOException e) {
                 log.warn("Failed to delete temp directory: {}", directory.toString(), e);

--- a/app/common/src/main/java/stirling/software/common/util/TempFileRegistry.java
+++ b/app/common/src/main/java/stirling/software/common/util/TempFileRegistry.java
@@ -86,6 +86,18 @@ public class TempFileRegistry {
     }
 
     /**
+     * Unregister a temporary directory from the registry.
+     *
+     * @param directory The directory to unregister
+     */
+    public void unregisterDirectory(Path directory) {
+        if (directory != null) {
+            tempDirectories.remove(directory);
+            log.debug("Unregistered temp directory: {}", directory.toString());
+        }
+    }
+
+    /**
      * Register a third-party temporary file that requires special handling.
      *
      * @param file The third-party temp file

--- a/app/common/src/test/java/stirling/software/common/service/TempFileCleanupServiceMoreTest.java
+++ b/app/common/src/test/java/stirling/software/common/service/TempFileCleanupServiceMoreTest.java
@@ -176,11 +176,13 @@ class TempFileCleanupServiceMoreTest {
     class ScheduledCleanup {
 
         @Test
-        @DisplayName("deletes registered temp directories and reports counts")
+        @DisplayName("deletes stale registered temp directories and reports counts")
         void deletesRegisteredDirectories() throws IOException {
             when(tempFileManager.cleanupOldTempFiles(anyLong())).thenReturn(2);
             Path regDir = Files.createDirectories(tempDir.resolve("registeredDir"));
             Files.createFile(regDir.resolve("inside.txt"));
+            Files.setLastModifiedTime(
+                    regDir, FileTime.fromMillis(System.currentTimeMillis() - 2L * 60 * 60 * 1000));
             Set<Path> dirs = new HashSet<>();
             dirs.add(regDir);
             when(registry.getTempDirectories()).thenReturn(dirs);
@@ -191,6 +193,22 @@ class TempFileCleanupServiceMoreTest {
             // The registered directory was removed by GeneralUtils.deleteDirectory.
             assertThat(Files.exists(regDir)).isFalse();
             verify(tempFileManager).cleanupOldTempFiles(anyLong());
+        }
+
+        @Test
+        @DisplayName("keeps a fresh registered temp directory")
+        void keepsFreshRegisteredDirectory() throws IOException {
+            when(tempFileManager.cleanupOldTempFiles(anyLong())).thenReturn(0);
+            Path regDir = Files.createDirectories(tempDir.resolve("freshRegisteredDir"));
+            Files.createFile(regDir.resolve("inside.txt"));
+            Set<Path> dirs = new HashSet<>();
+            dirs.add(regDir);
+            when(registry.getTempDirectories()).thenReturn(dirs);
+            lenient().when(registry.contains(any(File.class))).thenReturn(false);
+
+            withIsolatedUserHome(cleanupService::scheduledCleanup);
+
+            assertThat(Files.exists(regDir)).isTrue();
         }
 
         @Test

